backend/accounts/baseaccount.go

```diff
@@ -173,3 +178,62 @@ func (account *BaseAccount) SetTxNote(txID string, note string) error {
 	account.config.OnEvent(EventStatusChanged)
 	return nil
 }
+
+// ExportCSV implements accounts.Account.
+func (account *BaseAccount) ExportCSV(transactions []Transaction, w io.Writer) error {
+	writer := csv.NewWriter(w)
+	defer writer.Flush()
+	err := writer.Write([]string{
+		"Time",
+		"Type",
+		"Amount",
+		"Unit",
+		"Fee",
+		"Address",
+		"Transaction ID",
+		"Note",
+	})
+	if err != nil {
+		return errp.WithStack(err)
+	}
+
+	for _, transaction := range transactions {
+		transactionType := map[TxType]string{
+			TxTypeReceive:  "received",
+			TxTypeSend:     "sent",
+			TxTypeSendSelf: "sent_to_yourself",
+		}[transaction.Type()]
+		feeString := ""
+		fee := transaction.Fee()
+		if fee != nil {
+			feeString = fee.BigInt().String()
+		}
+		unit := account.Coin().SmallestUnit()
+		timeString := ""
+		if transaction.Timestamp() != nil {
+			timeString = transaction.Timestamp().Format(time.RFC3339)
+		}
+		for _, addressAndAmount := range transaction.Addresses() {
+			if transactionType == "sent" && addressAndAmount.Ours {
+				transactionType = "sent_to_yourself"
+			}
+			err := writer.Write([]string{
+				timeString,
+				transactionType,
+				addressAndAmount.Amount.BigInt().String(),
+				unit,
+				feeString,
+				addressAndAmount.Address,
+				transaction.TxID(),
+				account.Notes().TxNote(transaction.InternalID()),
+			})
+			if err != nil {
+				return errp.WithStack(err)
+			}
+			// a multitx is output in one row per receive address. Show the tx fee only in the
+			// first row.
+			feeString = ""
+		}
+	}
+	return nil
```

Comment:
Better check for flush error. It may actually fail. I'd flush here at the very end:

```go
writer.Flush()
return writer.Error()
```

> Error reports any error that has occurred during a previous Write or Flush. 

https://golang.org/pkg/encoding/csv/#Writer.Error

Reply:
Done